core/Session.php

```diff
@@ -78,8 +81,8 @@ public static function start($options = false)
         @ini_set('session.referer_check', '');
 
         // to preserve previous behavior piwik_auth provided when it contained a token_auth, we ensure
-        // the session data won't be deleted until the cookie expires.
-        @ini_set('session.gc_maxlifetime', $config->General['login_cookie_expire']);
+        // the session data won't be deleted until the cookie expires (if form rememberme is used).
+        @ini_set('session.gc_maxlifetime', Login::isRememberMeLogin() ? $config->General['login_cookie_expire'] : self::NON_REMEMBERME_GC_SESSION);
```

Comment:
I'm not sure about this one @diosmosis I  would have expected to always have it set to login cookie expire. 

Reading in http://php.net/manual/en/session.configuration.php#ini.session.gc-maxlifetime I'm not sure I understand it. Most of the time, `Login::isRememberMeLogin()` isn't true so it would maybe destroy the session too early? Also does this setting affect other sessions  or only the current session?

Is this change really needed?

Reply:
I was trying to make sure that the session data is removed after a reasonable time when session is not remembered, but I'm not sure there is a good way to do this. I'll just remove this code.